Tests/SwiftParserTest/Assertions.swift

```diff
@@ -156,6 +226,18 @@ func AssertDiagnostic<T: SyntaxProtocol>(
   if let highlight = spec.highlight {
     AssertStringsEqualWithDiff(diag.highlights.map(\.description).joined(), highlight, file: file, line: line)
   }
+  if let notes = spec.notes {
+    if diag.notes.count != notes.count {
+      XCTFail("""
+      Expected \(notes.count) notes but received \(diag.notes.count):
```

Comment:
Could print the expected notes as well?

Reply:
I explicitly didn’t print the expected notes because the test failure will point to where the `DiagnosticSpec` is declared and the expected notes will be shown there anyway. So I think there is very little value in printing the expected notes. This matches what we are doing for diagnostics as well.

Reply:
I had a comment there too :P. I generally like having everything in the error, makes it easier to attach them to issues and figure out the problem at a glance (rather than going to the test, comparing the output to what's in the test, etc). I don't feel particularly strongly about this though.

Reply:
I disagree but fair enough. If you want to add it, I won’t oppose you. 😜